### ISSUE TITLE
Apply labels based on author

### ISSUE THREAD

**Issue by damithc** [nus-cs2103/cs2103-pr-bot#19]
Given a csv file containing a table of github user IDs and a list of labels, it would be nice if the bot can apply those labels. 
e.g.

Github ID | team label | tutorial label | tutor | reviewer
---- | ---- | ---- | ---- | ----
johndoe | team.A1 | tutorial.W09 | tom2010 | tom2010
maryho | team.A2 | tutorial.W09 | tom2010 | jacobLee

Note that in some cases tutor and reviewer could be different.

**Comment by louislai** [comment]
Are there other types of labels that you have in mind beside team label and tutorial labels? If it is just these labels, it seems to me that the bot can just grab them from the PR title, without putting them in a csv file explicitly.

**Comment by damithc** [comment]
> Are there other types of labels that you have in mind beside team label and tutorial labels? If it is just these labels, it seems to me that the bot can just grab them from the PR title, without putting them in a csv file explicitly.

The idea is, when the bot can do this, the PR title need not have tutorial/team ID at all. Less work for students and fewer chances for students to make mistakes.

**Comment by louislai** [comment]
Okay i did not think of that. That is definitely less error-prone. 

By the way, regarding the format of the csv file, would it better for it to something like this 

Github ID | labels | tutor | reviewer
---- | ---- | ---- | ---- 
johndoe | team.A1, tutorial.W09 | tom2010 | tom2010
maryho | team.A2, tutorial.W09 | tom2010 | jacobLee

where labels are delimited by commas, so that if in the future there are other types of labels, those can be handled as well. I am not sure if we really need to know which label is the team label and which is tutorial label.

**Comment by damithc** [comment]
> By the way, regarding the format of the csv file, would it better for it to something like this

That's a good point. If you consider easy editability of the csv file (using excel), perhaps it is better to support something like this where any column with the `label` heading is taken as a label. that way, we can have any number of labels. 

Github ID | label | label | tutor | reviewer
---- | ---- | ---- | ---- | ----
johndoe | team.A1 | tutorial.W09 | tom2010 | tom2010
maryho | team.A2 | tutorial.W09 | tom2010 | jacobLee

Alternatively, we can omit the header row completely and consider all columns starting from the 3rd one are labels.

**Comment by louislai** [comment]
> Alternatively, we can omit the header row completely and consider all columns starting from the 3rd one are labels.

Should this be starting from the __4th__ column, since the first 3 are `Github ID`, `tutor` and `reviewer`?

If I have to choose between the two options, I would prefer keeping the header row and using the `label` heading as it is more flexible (columns can be in any order) and clearer to future maintainers (it's clear which column denotes which item without having to ask someone else what is the convention)

**Comment by damithc** [comment]
> If I have to choose between the two options, I would prefer keeping the header row

👍 

> Should this be starting from the 4th column

Yes. In fact, n<sup>th</sup> column, as we might add more non-label columns later.

**Comment by madsonic** [comment]
I was just wondering is the `Team label` unique? and what purpose does the `Tutorial label` serve?  Because if uniquely identify a team then we would know which tutorial they belong to.

**Comment by damithc** [comment]
Yes, the team ID is unique. However, a tutorial label is useful when you want to filter by tutorial. e.g. `label:tutorial.W09`
From the perspective of the pr-bot, it shouldn't matter. Just apply whatever label given in the csv file :-)
